### PR TITLE
NIAD-1154:GPCC Set Environment Variables

### PR DESCRIPTION
**Pull request by RIashfaqahmed**

### CODE REVIEW

File: terraform/aws/components/gp2gp/gpc-consumer-variables.tf [hunk 1]
```diff
@@ -189,4 +201,10 @@ variable secret_name_spine_sub_ca_cert {
 variable secret_name_sds_apikey {
   type = string
   default = "opentest-client-key"
-}
\ No newline at end of file
+}
+
+variable "gpc-consumer_empty_string" {
```
Comment (mattd-kainos) [inline terraform/aws/components/gp2gp/gpc-consumer-variables.tf]: What is this variable? It is actually needed?

File: terraform/aws/etc/eu-west-2_build1.tfvars [hunk 2]
```diff
@@ -50,18 +50,13 @@ gpc-consumer_service_container_port = 8080
 gpc-consumer_service_launch_type = "FARGATE"
 gpc-consumer_root_log_level = "WARN"
 gpc-consumer_log_level = "INFO"
-gpc-consumer_logging_format = "(*)"
-gpc-consumer_mesh_host = "https://msg.opentest.hscic.gov.uk/messageexchange/"
-gpc-consumer_mesh_cert_validation = "true"
-gpc-consumer_mongo_options = "replicaSet=rs0&readPreference=secondaryPreferred&retryWrites=false"
-gpc-consumer_ssl_trust_store_url = "s3://nhsd-aws-truststore/rds-truststore.jks"
+gpc-consumer_logging_format = ""
```
Comment (mattd-kainos) [inline terraform/aws/etc/eu-west-2_build1.tfvars]: this variable should be removed

File: terraform/aws/etc/eu-west-2_ptl.tfvars [hunk 3]
```diff
@@ -88,20 +88,15 @@ gpc-consumer_service_container_port = 8080
 gpc-consumer_service_launch_type = "FARGATE"
 gpc-consumer_root_log_level = "WARN"
 gpc-consumer_log_level = "INFO"
-gpc-consumer_logging_format = "(*)"
-gpc-consumer_mesh_host = "https://msg.int.spine2.ncrs.nhs.uk/messageexchange/"
-gpc-consumer_mesh_cert_validation = "true"
-gpc-consumer_mongo_options = "replicaSet=rs0&readPreference=secondaryPreferred&retryWrites=false"
-gpc-consumer_ssl_trust_store_url = "s3://nhsd-aws-truststore/rds-truststore.jks"
+gpc-consumer_logging_format = ""
```
Comment (mattd-kainos) [inline terraform/aws/etc/eu-west-2_ptl.tfvars]: remove

File: terraform/aws/etc/eu-west-2_vp.tfvars [hunk 4]
```diff
@@ -66,18 +66,13 @@ gpc-consumer_service_container_port = 8080
 gpc-consumer_service_launch_type = "FARGATE"
 gpc-consumer_root_log_level = "WARN"
 gpc-consumer_log_level = "INFO"
-gpc-consumer_logging_format = "(*)"
-gpc-consumer_mesh_host = "hhttps://mesh.vp.nhsredteam.internal.nhs.uk:8829/messageexchange/"
-gpc-consumer_mesh_cert_validation = "true"
-gpc-consumer_mongo_options = "replicaSet=rs0&readPreference=secondaryPreferred&retryWrites=false"
-gpc-consumer_ssl_trust_store_url = "s3://nhsd-aws-truststore/rds-truststore.jks"
+gpc-consumer_logging_format = ""
```
Comment (mattd-kainos) [inline terraform/aws/etc/eu-west-2_vp.tfvars]: remove

File: terraform/aws/etc/eu-west-2_vp.tfvars [hunk 5]
```diff
@@ -66,18 +66,13 @@ gpc-consumer_service_container_port = 8080
...
 gpc-consumer_logs_datetime_format = "%Y-%m-%d %H:%M:%S%L"
-gpc-consumer_create_wiremock = false
-gpc-consumer_wiremock_container_port = 8080
-gpc-consumer_wiremock_application_port = 8080
 gpc-consumer_sds_url = "https://sandbox.api.service.nhs.uk/spine-directory/"
```
Comment (mattd-kainos) [inline terraform/aws/etc/eu-west-2_vp.tfvars:70]: `gpc-consumer_sds_url` should be wiremock, no sandbox for vp

Comment (mattd-kainos) [reply terraform/aws/etc/eu-west-2_vp.tfvars:70]: maybe just set to empty string for now and we'll pick it up again closer to running vp tests. I just don't want us to run the tests against sandbox accidentally

File: terraform/aws/components/gp2gp/gpc-consumer-variables.tf (hunk 1 above)
Comment (RIashfaqahmed) [reply terraform/aws/components/gp2gp/gpc-consumer-variables.tf]: I defined this var for testing purposes to populate empty string to gpcc secret certs, but data secrets gave error on passing empty strings. It is removed now.

File: terraform/aws/etc/eu-west-2_build1.tfvars (hunk 2 above)
Comment (RIashfaqahmed) [reply terraform/aws/etc/eu-west-2_build1.tfvars]: Removed

Comment (RIashfaqahmed) [reply terraform/aws/etc/eu-west-2_build1.tfvars:53]: Removed

File: terraform/aws/etc/eu-west-2_ptl.tfvars (hunk 3 above)
Comment (RIashfaqahmed) [reply terraform/aws/etc/eu-west-2_ptl.tfvars]: Removed

File: terraform/aws/etc/eu-west-2_vp.tfvars (hunk 4 above)
Comment (RIashfaqahmed) [reply terraform/aws/etc/eu-west-2_vp.tfvars]: Removed

File: terraform/aws/etc/eu-west-2_vp.tfvars (hunk 5 above)
Comment (RIashfaqahmed) [reply terraform/aws/etc/eu-west-2_vp.tfvars:70]: Its a default value to be used if GP2GP-Wiremock is not enabled in environment. It gets changed to wiremock URL if wiremock is enabled. see following;

[https://github.com/nhsconnect/integration-adaptors/blob/0a2da08a2d25136e61a1f9b9d47e67c10727cb7e/terraform/aws/components/gp2gp/gpc-consumer-locals_env_variables.tf#L24](https://github.com/nhsconnect/integration-adaptors/blob/0a2da08a2d25136e61a1f9b9d47e67c10727cb7e/terraform/aws/components/gp2gp/gpc-consumer-locals_env_variables.tf#L24)

**Review by mattd-kainos — APPROVED** [review]

**Pull request merged**